dsp-shared/src/main/scala/dsp/valueobjects/Schema.scala

```diff
@@ -0,0 +1,209 @@
+        return Validation.fail(
+          BadRequestException(
+            s"Duplicate gui attributes for salsah-gui:guiElement $guiElement."
+          )
+        )
+      }
+
+      // If the gui element is a list, radio, pulldown or slider, check if a gui attribute (which is mandatory in these cases) is provided
+      val guiElementList     = SalsahGui.List
+      val guiElementRadio    = SalsahGui.Radio
+      val guiElementPulldown = SalsahGui.Pulldown
+      val guiElementSlider   = SalsahGui.Slider
+
+      val guiElementsPointingToList: Set[SalsahGui.IRI] = Set(guiElementList, guiElementRadio, guiElementPulldown)
+
+      val needsGuiAttribute: Boolean = guiElement match {
+        case None => false
+        case Some(guiElement) =>
+          guiElement.value == guiElementSlider ||
+            guiElementsPointingToList.contains(guiElement.value)
+      }
+
+      if (needsGuiAttribute) {
+        if (guiAttributes.isEmpty) {
+          return Validation.fail(BadRequestException(SchemaErrorMessages.GuiAttributesMissing))
+        }
+        val validatedGuiAttributes: Validation[Throwable, List[GuiAttribute]] = guiElement match {
+          // gui element is a list, radio or pulldown, so it needs a gui attribute that points to a list
+          case Some(guiElement) if guiElementsPointingToList.contains(guiElement.value) =>
+            validateGuiObjectsPointingToList(guiElement, guiAttributes).fold(
+              e => Validation.fail(e.head),
+              v => Validation.succeed(v)
+            )
+
+          // gui element is a slider, so it needs two gui attributes min and max
+          case Some(guiElement) if guiElement.value == guiElementSlider =>
+            validateGuiObjectSlider(guiElement, guiAttributes).fold(
+              e => Validation.fail(e.head),
+              v => Validation.succeed(v)
+            )
+
+          case _ =>
+            Validation.fail(
+              BadRequestException(
+                s"Unable to validate gui attributes. Unknown value for salsah-gui:guiElement: $guiElement."
+              )
+            )
+        }
+
+        return validatedGuiAttributes.fold(
+          e => Validation.fail(e.head),
+          v => Validation.succeed(new GuiObject(v, guiElement) {})
+        )
+
+      }
+
+      Validation.succeed(new GuiObject(guiAttributes, guiElement) {})
+    }
+  }
+
+  /**
+   * GuiAttribute value object.
+   *
+   * @param k the key parameter of the gui attribute ('min', 'hlist', 'size' etc.)
+   * @param v the value parameter of the gui attribute (an int, a list's IRI etc.)
+   */
+  sealed abstract case class GuiAttribute private (k: String, v: String) {
+    val value = k + "=" + v
+  }
+  object GuiAttribute {
+    def make(keyValue: String): Validation[Throwable, GuiAttribute] = {
+      val k: String = keyValue.split("=").head.trim()
+      val v: String =
+        keyValue.split("=").last.trim() // TODO also check the type of the value (integer, string etc.)
```

Comment:
Why not to do in this PR?

Reply:
I do this task in https://dasch.atlassian.net/browse/DEV-1082